docusaurus/docs/introduction/plugin-lifecycle.md

```diff
@@ -0,0 +1,89 @@
+
+For [backend](./backend.md) plugins, there is an additional initialization process (see [Phase 4](#phase-4-backend-plugin-initialization)).
+
+### Phase 1. Plugin discovery 
+
+Grafana starts to discover which plugins are installed by scanning directories on the file system for every `plugin.json`.
+
+### Phase 2. Plugin loading 
+
+All plugins that were discovered in the discovery phase are checked to make sure they’re valid. Some of the automated checks include:
+- Plugins must have a valid [signature](https://grafana.com/docs/grafana/latest/administration/plugin-management/#plugin-signatures). Valid plugins are referred to as _verified plugins_. 
+- Angular detection: Given that [Angular is deprecated](https://grafana.com/docs/grafana/latest/developers/angular_deprecation/), if Angular support is disabled and Angular is detected in the plugin, then we record an error and don't allow the plugin to be loaded.
+
+### Phase 3. Backend plugin initialization
+
+For any verified plugin that has a backend, Grafana configures the backend client to use HashiCorp’s Go Plugin System over RPC.
+
+### Phase 4. Registration
+
+All verified plugins are registered in an in-memory registry. From now on, the plugin is available within Grafana and so are referred to as _registered plugins_.
+
+Registered plugins show as installed in the catalog and appear in views for selecting a panel or data source in a dashboard.
+
+### Phase 5. Start backend plugin
+
+For registered plugins that have a backend, Grafana starts to run the backend binary as a separate process using HashiCorp’s Go Plugin System over RPC. The supported plugin protocol and version is negotiated between Grafana (client) and the plugin (server) to give Grafana an understanding of the plugin's capabilities. 
+
+A Grafana backend plugin has its own separate life cycle. So long as the backend plugin is running, Grafana will make sure to restart the backend plugin in case it crashes or is terminated. When Grafana is shut down, the backend processes is then terminated.  
+
+### Phase 6. Client-side loading 
+
+After Grafana has started and the HTTP API is running, Grafana users receive the server-side rendered index page containing so-called bootstrap data. This data includes the list of available plugins and a URI to a `module.js` file that Grafana uses to instantiate the plugin. 
```

Comment:
Nit: we could link to https://grafana.com/docs/grafana/latest/developers/http_api/

Reply:
done.